File: job-manager/src/types/avatar.ts
import { Timestamp } from 'firebase-admin/firestore';

export enum AvatarGender {
  male = 'male',
  female = 'female'
}

export enum AvatarTypes {
  twin = 'twin',
  synthetic = 'synthetic'
}

export type AvatarParameters = {
  gender: AvatarGender;
  ethnicity: string;
  skinColor: string;
  age: string;
  attractiveness: string;
  body: string;
  face: string;
  hairStyle: string;
  hairColor: string;
  eyes: string;
  skin: string;
  facialHair: string;
  nose: string;
  eyeLashes: string;
  eyeBrows: string;
  outfit: string;
  lips: string;
  bustSize: string;
  ears: string;
  bodyHair: string;
  height: string;
}

export type LoraConfig = {
  path: string;
  filename: string;
}

export type AvatarLoras = {
  qwenEdit2511: LoraConfig;
  wan22T2vA14b?: LoraConfig;
}

export type Avatar = {
  id?: string;
  userId?: string;
  name: string;
  slug: string;
  type: AvatarTypes;
  parameters: AvatarParameters;
  mainImagePath?: string;
  voicePath?: string;
  isUploadedVoice?: boolean;
  photoSetGenerated: boolean;
  loras: AvatarLoras;
  createdAt?: Timestamp;
  updatedAt?: Timestamp;
}

File: shared-package-py/loom24_shared/types/avatar.py
from __future__ import annotations

from enum import Enum
from typing import Any, Optional

from pydantic import BaseModel


class AvatarGender(str, Enum):
    male   = 'male'
    female = 'female'


class AvatarTypes(str, Enum):
    twin      = 'twin'
    synthetic = 'synthetic'


class AvatarParameters(BaseModel):
    gender:        AvatarGender
    ethnicity:     str
    skinColor:     str
    age:           str
    attractiveness: str
    body:          str
    face:          str
    hairStyle:     str
    hairColor:     str
    eyes:          str
    skin:          str
    facialHair:    str
    nose:          str
    eyeLashes:     str
    eyeBrows:      str
    lips:          str
    bustSize:      str
    ears:          str
    bodyHair:      str
    height:        str


class Avatar(BaseModel):
    id:            Optional[str] = None
    userId:        Optional[str] = None
    name:          str
    slug:          str
    type:          AvatarTypes
    parameters:    AvatarParameters
    mainImagePath: Optional[str] = None
    voiceId:       Optional[str] = None
    createdAt:     Optional[Any] = None
    updatedAt:     Optional[Any] = None
